### DISCUSSION TITLE
discord provider user id problem

### DISCUSSION THREAD

**Discussion by ghost** [nextauthjs/next-auth#3747] [Help]
### Question 💬

how can I get the user id for the discord provider

### How to reproduce ☕️

```
import { getSession } from "next-auth/react"

export default async (req, res) => {
  const session = await getSession({ req })

  if (session) {
    res.send({
      user: {
        id: session.user.image,
        name: session.user.name,
        email: session.user.email,
        image: session.user.image,
      },
    })
  } else {
    res.send({
      error: "You must be sign in to view the protected content on this page.",
    })
  }
}
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by ryanbarr** [comment]
Hey @UmutKDev,

To make sure I understand the problem: you're trying to get either the user's ID or the user's Discord ID and return it in the response in this API call.

By default, the session returned only contains these [default Session properties](https://github.com/nextauthjs/next-auth/blob/8ff4b260143458c5d8a16b80b11d1b93baa0690f/types/index.d.ts#L437-L444). If you'd like to add additional things to the session object, you can do so by following the [instructions to add a custom session callback](https://next-auth.js.org/configuration/callbacks#session-callback).
